### ISSUE TITLE
[UI] Selecting audio / text tracks by name insteat of language when multiple same language tracks present in HLS

### ISSUE THREAD

**Issue by mindaugasvcs** [shaka-project/shaka-player#8300]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes

**If the problem is related to FairPlay, have you read the tutorial?**
Not related

**What version of Shaka Player are you using?**
The latest one at the moment 4.14.2

**Can you reproduce the issue with our latest release version?**
Yes

**Can you reproduce the issue with the latest code from `main`?**
I bet I could as this issue is not fixed for many years

**Are you using the demo app or your own custom app?**
https://ajax.googleapis.com/ajax/libs/shaka-player/4.14.2/shaka-player.ui.js

**If custom app, can you reproduce the issue using our demo app?**

**What browser and OS are you using?**
Brave, Windows 10

**For embedded devices (smart TVs, etc.), what model and firmware version are you using?**

**What are the manifest and license server URIs?**

**What configuration are you using?  What is the output of `player.getNonDefaultConfiguration()`?**
`player.configure("abr.restrictToElementSize", true);`

**What did you do?**
In the UI tried to change audio / text track

**What did you expect to happen?**
See full list of available tracks no matter if language is the same

**What actually happened?**
The list contains only some tracks, the label is guessed from language metadata in HLS. Instead the label must be taken from the name metadata in HLS.

**Are you planning to send a PR to fix it?**
Yes when I know the inner workings of this lib.

**Comment by absidue** [comment]
All you need to do is tell the UI in the configuration that you want it to use the labels instead of the language, by setting `trackLabelFormat: shaka.ui.Overlay.TrackLabelFormat.LABEL` in the UI configuration (it is separate from the player configuration).

**Comment by avelad** [comment]
@absidue is right, so this is not an issue.

**Comment by mindaugasvcs** [comment]
> [@absidue](https://github.com/absidue) is right, so this is not an issue.

Why so quick on closing it? shaka.ui.Overlay.TrackLabelFormat.LABEL (value 3) is the only working option, the other options (0, 1, 2) are buggy: not showing full list of tracks. Also what is the purpose of ROLE? HLS manifest files don't have it. Default should be LABEL_LANGUAGE.

**Comment by mindaugasvcs** [comment]
> shaka-player supports more than just HLS manifests. 1 and 2 work are meant to be used with DASH manifests, which do have roles. 0 works correctly when you have one audio track per language, if that is not your use case you can easily change the config to 3 for the labels.
> 
> LANGUAGE_ROLE is not a fallbacking thing, it creates the display label as label + role (e.g. "Spanish dub"), so your proposed 5th option of LABEL_LANGUAGE would be label + language (e.g. "Spanish Spanish").

Ok, despite shaka-player knows then it plays HLS and automatically runs code for HLS playback without any manual configuration for HLS playback but for some unknown reasons the configuration of labels is not automatic? I'm just missing some consistency or common sence here. Also this explanation does not say nothing why it does not show full list of tracks for options 0, 1, 2? I won't argue if it's a bug or not but if you noticed LANGUAGE is not even required in HLS manifest files, so the default option for HLS is kind of invalid.

**Comment by avelad** [comment]
Do you have a sample stream to try to implement something in Shaka Player?

**Comment by absidue** [comment]
> Also this explanation does not say nothing why it does not show full list of tracks for options 0, 1, 2?

Because shaka-player deduplicates the displayed list of tracks based on your chosen display label, as showing users multiple tracks with identical display labels is unhelpful. That means if your config is telling shaka-player to use the language as the display label, if you have multiple audio tracks with the language Spanish it will only list one of them.

I really don't understand why you are still arguing about this, it's one simple easy to change setting on your side, but instead you are asking the shaka-player developers to implement some kind of auto-detection, which would be a breaking change.

**Comment by mindaugasvcs** [comment]
This possible BC is your fault when deciding to also implement HLS format into one lib while not properly evaluating the differences between the two formats.
Making assumptions on what end users would see as helpful or not is also your fault as it's just data that player must show.
Of course, you don't have to take my constructive criticism into account, but calling it an argument is a bit too much. Just tried to point out some flaws.

**Comment by avelad** [comment]
@mindaugasvcs I'm willing to implement something in Shaka Player, but for that you need to answer the question:

```
**What are the manifest and license server URIs?**
<!-- NOTE:
  You can send the URIs to <shaka-player-maintainers@googlegroups.com> instead,
  but please use GitHub and the template for the rest.
  A copy of the manifest text or an attached manifest will **not** be
  enough to reproduce your issue, and we **will** ask you to send a
  URI instead.  You can copy the URI of the demo app to send us the
  exact asset, licence server, and settings you have selected there.
  Avoid geo-blocked content if at all possible, as the contributors who diagnose
  Shaka Player issues may not be able to access it.
  If you send the URIs to email, the response time and resolution
  will be much higher.
-->
```
